File: src/main/java/com/baulsupp/oksocial/output/OsxOutputHandler.java
package com.baulsupp.oksocial.output;

import java.io.IOException;

import static java.util.Arrays.asList;
import static java.util.Optional.of;

public class OsxOutputHandler<R> extends ConsoleHandler<R> {
  public OsxOutputHandler(ResponseExtractor<R> responseExtractor) {
    super(responseExtractor);
  }

  public void openPreview(R response) throws IOException {
    streamToCommand(of(responseExtractor.source(response)),
        asList("open", "-f", "-a", "/Applications/Preview.app"), of(30));
  }
}

File: src/main/java/com/baulsupp/oksocial/output/OsxOutputHandler.kt
package com.baulsupp.oksocial.output

import java.io.IOException

import java.util.Arrays.asList

open class OsxOutputHandler<R>(responseExtractor: ResponseExtractor<R>) : ConsoleHandler<R>(responseExtractor) {

    @Throws(IOException::class)
    override fun openPreview(response: R) {
        streamToCommand(responseExtractor.source(response),
                asList("open", "-f", "-a", "/Applications/Preview.app"), 30)
    }
}
